### PR TITLE
IDE-2054 workspace can have only one sdk

### PR DESCRIPTION
**Pull request by simonjhy**

### CODE REVIEW

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java [hunk 1]
```diff
@@ -119,34 +156,25 @@ public void resourceChanged( final IResourceChangeEvent event )
         try
         {
             final IResourceDelta delta = event.getDelta();
-            final SDK sdk = SDKUtil.getWorkspaceSDK();
 
-            if( delta != null && sdk != null )
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java]: why did you remove teh null check?  we had to add it previously?

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java [hunk 2]
```diff
@@ -119,34 +156,25 @@ public void resourceChanged( final IResourceChangeEvent event )
...
+            if( delta != null )
             {
                 for( IResourceDelta child : delta.getAffectedChildren() )
                 {
-                    final IPath deltaLocation = child.getResource().getLocation();
+                    final IResourceDelta[] sdkChangedFiles =
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java]: This code matches too much.  What if we have a project that is in not way related to the SDK or SDK workspace project and it has a file called "build.properties" in it.  this pattern matcher is going ot match isn't it?

Please be careful to think of other situations that don't exactly match the specific thing you are coding for and see if you can break your own code assumptions.  

You just named your variable "sdkChangedFiles"  but you don't know that... the files could be any resource in the workspace that has changed... like "myFooProject" and it in it has a "myFooProject/build.properties" file.  That isn't an "sdkChangedFile".

**Comment by gamerson** [comment]
hey @simonjhy you made some incorrect assumptions in part of this file, please check it again and resend. thanks.

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java (hunk 2 above)
Comment (simonjhy) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java]: because we merged the hasMoreThanOneSdkListener into sdkBuildProertiesListener, we need to handle the event that sdk project can be deleted. While this happened, the child.getResource will be return null and it will throw null exception, so I removed the sdk null checking. I want to make the resourceChange method can be mapped to two different listener check.

Comment (simonjhy) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java]: you mean we still need to identify the delta chagne belong to the sdk project?

Comment (simonjhy) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java]: whiile the sdk project has been deleted, the sdkBuildPropertiesListener monitor the  post event, so we can't get sdk instance in that moment. I will try to change it.
